roboteam_ai/src/skills/AvoidBallForBallPlacement.cpp

```diff
@@ -9,80 +9,67 @@
 namespace rtt {
 namespace ai {
 
-AvoidBallForBallPlacement::AvoidBallForBallPlacement(std::string name, bt::Blackboard::Ptr blackboard)
-: Skill(std::move(name), std::move(blackboard)) {}
-
-void AvoidBallForBallPlacement::onInitialize() {
-    ballPlacementTargetLocation = interface::InterfaceValues::getBallPlacementTarget();
 
-    auto robotPos = rtt::Vector2(robot->pos);
-
-    if (positionIsTooCloseToLine(robotPos)) {
-        currentProgress = RUNNING;
-    } else {
-        currentProgress = DONE;
-        return;
-    }
-
-    auto ballPos = rtt::Vector2(ball->pos);
-    // get the target to move to which is the same vector from the ball
-    auto projectionOnLinePoint = robotPos.project(ballPos, ballPlacementTargetLocation);
-    auto lineDirection = ballPos - ballPlacementTargetLocation;
-    auto inversedLineDirectionX = Vector2(-lineDirection.x, lineDirection.y).stretchToLength(.5);
-    auto inversedLineDirectionY = Vector2(lineDirection.x, -lineDirection.y).stretchToLength(.5);
 
+AvoidBallForBallPlacement::AvoidBallForBallPlacement(std::string name, bt::Blackboard::Ptr blackboard)
+: Skill(std::move(name), std::move(blackboard)) { }
 
-    auto point = projectionOnLinePoint + inversedLineDirectionX;
-    if (Field::pointIsInField(point) && !positionIsTooCloseToLine(point)) {
-        targetToMoveTo = point;
-        return;
-    }
+bt::Node::Status AvoidBallForBallPlacement::onUpdate() {
+    auto robotPos = rtt::Vector2(robot->pos);
+    Vector2 force = {0, 0};
 
-    point = projectionOnLinePoint + inversedLineDirectionY;
-    if (Field::pointIsInField(point) && !positionIsTooCloseToLine(point)) {
-        targetToMoveTo = point;
-        return;
-    }
+    // forces from robots
+    for (auto otherRobot : World::getAllRobots()) {
+        double distance = robotPos.dist(otherRobot.pos);
+        Vector2 distanceVector = robotPos - otherRobot.pos;
 
-    // if we reach this point we are in a corner of the field.
-    inversedLineDirectionX.stretchToLength(2);
-    targetToMoveTo = control::ControlUtils::projectPositionToWithinField(projectionOnLinePoint + inversedLineDirectionX);
-    if (positionIsTooCloseToLine(targetToMoveTo)) {
-        inversedLineDirectionY.stretchToLength(2);
-        targetToMoveTo = control::ControlUtils::projectPositionToWithinField(projectionOnLinePoint + inversedLineDirectionY);
+        if (otherRobot.id != robot->id && distance < minRobotDistanceForForce) {
+            force = force + distanceVector.stretchToLength(1) * (robotWeight/(distance * distance));
```

Comment:
use:
force = force + distanceVector.normalize() * (robotWeight / distanceVector.length2())

Reply:
distanceVector.length2() appears weird to me. i refactored it to pow(vector, 2)